docs/design/CONFIG_MANAGEMENT.md

```diff
@@ -0,0 +1,261 @@
+Config Management Design
+===================
+
+## Motivation
+
+[Issue 43][issue_43] specifies the need for lifecycle management of Envoy Gateway system components that run as Go
+routines.
+
+## Goals
+
+* Gracefully manage the services that comprise Envoy Gateway.
+
+## Non-Goals
+
+* Provide a detailed Envoy Gateway configuration specification.
+
+## Proposal
+
+Introduce a `serve` argument to the `envoy-gateway` command. The `serve` argument encapsulates the management of Envoy
+Gateway.
+```go
+// gateway/internal/cmd/root.go
+
+package cmd
+
+import (
+	"github.com/spf13/cobra"
+
+	"github.com/envoyproxy/gateway/internal/cmd/serve"
+)
+
+// GetRootCommand returns the root cobra command to be executed
+// by main.
+func GetRootCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "envoy-gateway",
+		Short: "Manages Envoy Proxy as a standalone or Kubernetes-based application gateway",
+	}
+
+	cmd.AddCommand(serve.NewCommand())
+
+	return cmd
+}
+```
+
+The serve package provides the following functionality:
+- Allows the serve package to be run as a Cobra Command.
+- Validates, processes, etc. command line flags and the Envoy Gateway config file. 
+- Constructs an object, e.g. `GatewayServer`, based on the validated flags/config. This object encapsulates all the
+necessary parameters for running Envoy Gateway.
+- Runs Envoy Gateway based on the provided `GatewayServer`.
+
+A subset of the Envoy Gateway's configuration parameters may be set by a configuration file, as a substitute for
+command-line flags. Providing parameters via a config file is the recommended approach as it simplifies deployment
+and configuration management. The config file is defined by the `BootstrapConfig` struct. The config file must be a YAML
+representation of the parameters in this struct. For example:
+```yaml
+apiVersion: gateway.envoy.io/v1alpha1
+kind: BootstrapConfig
+foo:
+  bar: baz
+...
+```
+The `--config` flag specifies the path of the Envoy Gateway configuration file. Envoy Gateway will load its config from
```

Comment:
nit: prefer if we use existing envoy naming here https://www.envoyproxy.io/docs/envoy/latest/operations/cli#cmdoption-c

Reply:
@arkodg coomit [ea6dcd3](https://github.com/envoyproxy/gateway/pull/87/commits/ea6dcd316773e8c7e5cbc794480f9cc40c27dbca) resolves this feedback.